pkg/asset/machines/aws/worker.go

```diff
@@ -30,6 +37,83 @@ type MachineConfig struct {
 	Machine     types.AWSMachinePoolPlatform
 }
 
+func createWorkerMachineSets(config *WorkerConfig) metav1.List {
+	list := metav1.List{}
+	list.Kind = "List"
```

Comment:
It will not work. See https://golang.org/ref/spec#Struct_types where it states:

_Promoted fields act like ordinary fields of a struct except that they cannot be used as field names in composite literals of the struct._

We will have to do this for  in place literals:
```
list := metav1.List{TypeMeta: TypeMeta{Kind: "List"}}
```
which is less readable in my opinion.

Reply:
Ah, thanks for the links.  Personally, I prefer the literal with `TypeMeta`.  But I can see that going either way, and it's your PR, so I'm fine leaving this as you have it.